app/controllers/admin/user_creations_controller.rb

```diff
@@ -1,52 +1,75 @@
 class Admin::UserCreationsController < ApplicationController
   
   before_action :admin_only
+  before_action :get_creation
+  before_action :can_be_marked_as_spam, only: [:set_spam]
+
+  def get_creation
+    raise "Redshirt: Attempted to constantize invalid class initialize hide #{params[:creation_type]}" unless %w(ExternalWork Bookmark Work).include?(params[:creation_type])
```

Comment:
Curiosity more than anything: why exactly do we have "initialise hide #" in that message? (I remember the "Redshirt" was so we could find these messages in our logs)

Reply:
Ugh, copy and paste error. The Redshirt bit was repeated in several different actions, but this pull request moves it to a separate method... I evidently copy and pasted the Redshirt error from the hide action. Will remove!